test/suites/storage-version-migration.robot

```diff
@@ -0,0 +1,107 @@
+    ...    Validate Migration On Beta Version
+
+    Update Beta CRD To Stable
+
+    Wait Until Keyword Succeeds    5x    20s
+    ...    Restart MicroShift
+
+    Wait Until Keyword Succeeds    5x    10s
+    ...    Validate Migration On Stable Version
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Create initial setup with CRD and resources
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+
+    Create Beta CRD
+    Create Custom Resource
+
+Teardown
+    [Documentation]    Delete all created resources
+    Delete Migraiton Resources
+
+Create Beta CRD
+    [Documentation]    Create beta CRD
+    Run With Kubeconfig    oc apply -f ${BETA_CRD}
+
+Create Custom Resource
+    [Documentation]    Create beta version resource
+    Run With Kubeconfig    oc apply -f ${CR_RESOURCE}
+
+Update Beta CRD To Stable
+    [Documentation]    Update the beta versin of CRD to stable
+    Run With Kubeconfig    oc apply -f ${STABLE_CRD}
+
+Validate Migration On Beta Version
+    [Documentation]    Validate that a migration resource was created for the beta CRD,
+    ...    we should expect to see a StorageState and StorageVersionMigration resource created.
+    ...    With in the Migration and State objects we should see the correct version and object
+    ...    hash defined.
+
+    ${yaml_data}=    Oc Get    migrates.test.resource    ""    default
+    ${storage_yaml_data}=    Oc Get    storagestates.migration.k8s.io    ""    migrates.test.resource
+    ${migration_json_text}=    Run With Kubeconfig
+    ...    oc get storageversionmigration -o jsonpath="{.items[?(@.spec.resource.group=='test.resource')]}"
+    ${migration_json_data}=    Json Parse    ${migration_json_text}
+
+    Should Be Equal    ${yaml_data.apiVersion}    test.resource/v1beta1
+    Should Be Equal    ${storage_yaml_data.status.currentStorageVersionHash}    8F5KO5MYqcM=
+    Should Be Equal    ${migration_json_data.spec.resource.version}    v1beta1
+
+Validate Migration On Stable Version
+    [Documentation]    Validate that a migration resource was created for the stable CRD,
+    ...    we should expect to see the StorageState and StorageVersionMigration resource to be updated.
+    ...    With in the Migration and State objects we should see the correct version and object
+    ...    hash defined.
+
+    ${yaml_data}=    Oc Get    migrates.test.resource    ""    default
+    ${storage_yaml_data}=    Oc Get    storagestates.migration.k8s.io    ""    migrates.test.resource
+    ${migration_json_text}=    Run With Kubeconfig
+    ...    oc get storageversionmigration -o jsonpath="{.items[?(@.spec.resource.group=='test.resource')]}"
+    ${migration_json_data}=    Json Parse    ${migration_json_text}
+
+    Should Be Equal    ${yaml_data.apiVersion}    test.resource/v1
+    Should Be Equal    ${storage_yaml_data.status.currentStorageVersionHash}    O2TShlD54rQ=
+    Should Be Equal    ${migration_json_data.spec.resource.version}    v1
```

Comment:
These keywords are very similar, what do you think about following?

```suggestion
Validate Migration
    [Documentation]    Validate that a migration resource was created the CRD,
    ...    we should expect to see a StorageState and StorageVersionMigration resource created.
    ...    With in the Migration and State objects we should see the correct version and object
    ...    hash defined.
    [Arguments]     ${api_version}    ${hash}

    ${yaml_data}=    Oc Get    migrates.test.resource    ""    default
    ${storage_yaml_data}=    Oc Get    storagestates.migration.k8s.io    ""    migrates.test.resource
    ${migration_json_text}=    Run With Kubeconfig
    ...    oc get storageversionmigration -o jsonpath="{.items[?(@.spec.resource.group=='test.resource')]}"
    ${migration_json_data}=    Json Parse    ${migration_json_text}

    Should Be Equal    ${yaml_data.apiVersion}    test.resource/${api_version}
    Should Be Equal    ${storage_yaml_data.status.currentStorageVersionHash}    ${hash}
    Should Be Equal    ${migration_json_data.spec.resource.version}    ${api_version}
```

And use it like so:

```robot
  Wait Until Keyword Succeeds    5x    10s
    ...    Validate Migration    v1beta1    8F5KO5MYqcM=

(....)


  Wait Until Keyword Succeeds    5x    10s
    ...    Validate Migration    v1    O2TShlD54rQ=
```

Reply:
Oh that's a great idea, thanks!